pennylane_lightning/core/src/simulators/lightning_gpu/initSV.cu

```diff
@@ -95,6 +106,91 @@ void setStateVector_CUDA_call(GPUDataT *sv, index_type &num_indices,
                                                 indices);
     PL_CUDA_IS_SUCCESS(cudaGetLastError());
 }
+
+/**
+ * @brief The CUDA kernel that multiplies the state vector data on GPU device
+ * by a global phase.
+ *
+ * @param sv Complex data pointer of state vector on device.
+ * @param num_sv Number of state vector elements.
+ * @param phase Complex data pointer of input values (on device).
+ */
+template <class GPUDataT, class index_type>
+__global__ void globalPhaseStateVectorkernel(GPUDataT *sv, index_type num_sv,
+                                             GPUDataT phase) {
+    const unsigned int i = blockIdx.x * blockDim.x + threadIdx.x;
+    if (i < num_sv) {
+        sv[i] = ComplexMul(sv[i], phase);
+    }
+}
+
+/**
+ * @brief The CUDA kernel call wrapper.
+ *
+ * @param sv Complex data pointer of state vector on device.
+ * @param num_sv Number of state vector elements.
+ * @param phase Constant complex phase.
+ * @param thread_per_block Number of threads set per block.
+ * @param stream_id Stream id of CUDA calls
+ */
+template <class GPUDataT, class index_type>
+void globalPhaseStateVector_CUDA_call(GPUDataT *sv, index_type num_sv,
+                                      GPUDataT phase, size_t thread_per_block,
+                                      cudaStream_t stream_id) {
+    auto dv = std::div(static_cast<long>(num_sv), thread_per_block);
+    size_t num_blocks = dv.quot + (dv.rem == 0 ? 0 : 1);
+    const size_t block_per_grid = (num_blocks == 0 ? 1 : num_blocks);
+    dim3 blockSize(thread_per_block, 1, 1);
+    dim3 gridSize(block_per_grid, 1);
```

Comment:
Since these values are usually inferred from physical limits (as in, the total number of runnable threads, blocks, etc), are we sure the chosen numbers here, and arguments to the functions are sufficient to fit everything for all use-cases using this 1D indexing? Will we need to consider 2/3D blocks if things get larger?

Reply:
I think that since the data (and access pattern) is intrinsically 1D, CUDA will in effect linearize any 2D/3D mapping. We could even avoid using `dim3` here (copy-pasting) and elsewhere. The `<<<numBlocks, threadsPerBlock>>>` syntax should be sufficient, but maybe we'd lose the possibility to call this asynchronously with streams?

Reply:
Thanks @mlxd! Good question. Threads number per block is default as `256`, which is always less than the number of allowed threads per block as of `Ampere` card. `256` threads per block was chosen (which is mentioned in the following comment) is because this is the minimal number of threads per block (8 warps) required to hide maximal latency.